src/polyglot-notebooks-vscode-common/src/vscodeNotebookManagement.ts

```diff
@@ -31,13 +31,44 @@ export async function replaceNotebookCellMetadata(notebookUri: vscode.Uri, cellI
 }
 
 export async function replaceNotebookMetadata(notebookUri: vscode.Uri, documentMetadata: { [key: string]: any }): Promise<boolean> {
-    const notebookEdit = vscode.NotebookEdit.updateNotebookMetadata(documentMetadata);
-    const edit = new vscode.WorkspaceEdit();
-    edit.set(notebookUri, [notebookEdit]);
-    const succeeded = await vscode.workspace.applyEdit(edit);
-    return succeeded;
+    const notebook = vscode.workspace.notebookDocuments.find(d => d.uri === notebookUri);
+    if (notebook) {
+        const metadata = notebook.metadata;
+        const shouldUpdate = compare(metadata, documentMetadata);
```

Comment:
I was wondering why we need to update when the metadata has not changed i.e. when `compare` returns true. Could you please describe what the fix is doing here (just for my own understanding :))?

Reply:
changed that code